pkg/network/ebpf/c/protocols/http/http.h

```diff
@@ -109,7 +109,7 @@ static __always_inline http_transaction_t *http_fetch_state(http_transaction_t *
 }
 
 static __always_inline bool http_should_flush_previous_state(http_transaction_t *http, http_packet_t packet_type) {
-    return (packet_type == HTTP_REQUEST && http->request_started) ||
+    return (packet_type == HTTP_REQUEST && (http->request_started || http->response_status_code)) ||
```

Comment:
please add a comment that explains in words the state machine you have here, meaning what situations could occur in the real world traffic and how should we handle that. (preferably split each potential "state" in a separate bullet / line